File: QingYeMark/QingYeMarketparent/QingYeMarket_sellergoods_interface/src/main/java/com/qingyemarket/sellergoods/service/BrandService.java
package com.qingyemarket.sellergoods.service;

import com.qingyemarket.entity.PageResult;
import com.qingyemarket.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * @Author: qingye
 * @Date: 2019/4/10 0010 19:04
 * @Version 1.0
 */
public interface BrandService {

    /**
     * 查询所有品牌
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 返回分页列表
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize);

    /**
     * 添加品牌
     * @param tbBrand
     * @throws Exception
     */
    public void add(TbBrand tbBrand) throws Exception;

    /**
     * 修改品牌
     * @param tbBrand
     */

    void brandUpdate(TbBrand tbBrand) throws Exception;

    /**
     * 查询品牌
     * @param id
     * @return
     */
    TbBrand findOne(long id) ;

    /**
     * 根据id删除品牌
     * @param ids
     */
    void brandDelete(long[] ids) throws Exception;

    /**
     * 查询结果分页显示
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(TbBrand tbBrand,int pageNum,int pageSize);

    /**
     * 品牌下拉数据
     * @return
     */
    List<Map> selectOptionList();
}

File: QingYeMark/QingYeMarketparent/QingYeMarket_manager_web/src/main/webapp/js/service/BrandService.js
//自定义定义服务层
app.service("brandService", function ($http) {
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&rows=' + rows);
    };
    //查询单个品牌
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };
    //新增品牌
    this.save = function (entity) {
        return $http.post('../brand/brandAdd.do?', entity);
    };

    //修改品牌
    this.update = function (entity) {
        return $http.post('../brand/brandUpdate.do?', entity);
    };

    //删除品牌
    this.del = function (selectIds) {
        return $http.get('../brand/brandDelete.do?ids=' + selectIds);
    };

    this.search = function (page, rows, entity) {

        return $http.post('../brand/search.do?page=' + page + '&rows=' + rows, entity)
    }
    //下拉列表
    this.selectOptionList = function () {
        return  $http.get('../brand/selectOptionList.do');
    }
});